src/components/layouts/app-layout.tsx

```diff
@@ -60,6 +69,24 @@ const AppLayout: React.VoidFunctionComponent<Props> = ({ drawerContent, children
     setSyncingData(false);
   };
 
+  /**
+   * Handler for sync-date selection dialog
+   */
+  const handleClickOpen = () => {
+    if (!selectedStartDate) {
+      setSelectedStartDate(new Date(yesterday));
+    }
+    
```

Comment:
This check should not be in event handler

Reply:
Done. Moved to useEffect with a dependency of syncSelection.